### DISCUSSION TITLE
PWM Shift Register

### DISCUSSION THREAD

**Discussion by FotisDovas** [tttapa/Control-Surface#626] [Q&A]
Hi!
Is there PWM support for Shift Registers, from the Control Surface library?

**Comment by tttapa** [comment]
Like the [ShiftPWM](https://github.com/elcojacobs/ShiftPWM) library?

Not out of the box, though you could implement the ExtIO interface for ShiftPWM LEDs, or use ShiftPWM directly.  
That being said, you're probably better off using LED drivers or addressable LEDs that are designed for the purpose. (Although you have to be careful with clockless addressable LEDs, see https://github.com/FastLED/FastLED/wiki/Interrupt-problems.)

**Reply by FotisDovas** [reply]
Yeah, I implemented the shiftPWM library, and it works fine, with no major loads. 
 I was just wandering, just in case you had a better implementation yourself! 
The led tech you are referring to is unknown to me, but I'll check it!
Thank you for all your work man, all the features I had in mind are now ready, with minimal c++ knowledge by me. 
I swore that before the next microcontroller project, i will learn some syntax basics 😁
